merlin/schema/schema.py

```diff
@@ -239,13 +239,15 @@ def value_count(self) -> Optional[Domain]:
         value_count = self.properties.get("value_count")
         return Domain(**value_count) if value_count else None
 
-    def __merge__(self, other):
-        col_schema = self.with_tags(other.tags)
-        col_schema = col_schema.with_properties(other.properties)
-        col_schema = col_schema.with_dtype(
-            other.dtype, is_list=other.is_list, is_ragged=other.is_ragged
+    def __or__(self, other):
+        dtype = self.dtype if self.dtype != md.unknown else other.dtype
```

Comment:
dtype is currently the only part of self that overrides whats coming in.... Do we want the self to override what comes in from other in the rest of the attributes?

Reply:
I think if we're using `__or__` we want `self` to have priority over `other` across the board

Reply:
We do not want this. This actually breaks how we create ancestor schema in the dag. Unless we want to change how that works inside the merge logic in schema. This change alone causes breaks in the nvtabular tests. Particularly in the test_dask_nvt tests.

Reply:
the following line of code starts delivering the original dtype for the column instead of the latest dtype after transformations occur. https://github.com/NVIDIA-Merlin/core/blob/644d4e302f6cd27f6392cc9d862be7b18425cb1d/merlin/dag/node.py#L242

Reply:
Yeah, in order for `self` having priority to work, we'd have to flip the order of the arguments where it's used too. I think there's only one place we use `__merge__` though so hopefully that's not too bad?